From a186f3f767e2aeb47a9677194a749ec51707cda4 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Fri, 1 Aug 2025 13:26:34 +0200
Subject: [PATCH 09/55] kernel-headers: always build the v6.6 headers even for
 older kernels

Ensures kernel 6.6 headers are available even when building for older kernel versions, providing compatibility for packages requiring newer kernel APIs.

Signed-off-by: John Crispin <john@phrozen.org>
---
 toolchain/kernel-headers/Makefile | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/toolchain/kernel-headers/Makefile b/toolchain/kernel-headers/Makefile
index dc70f32c90..7e2d37ca25 100644
--- a/toolchain/kernel-headers/Makefile
+++ b/toolchain/kernel-headers/Makefile
@@ -14,6 +14,17 @@ override HOST_QUILT:=
 
 include $(INCLUDE_DIR)/kernel.mk
 
+# TIP
+KERNEL_PATCHVER := 6.6
+KERNEL_NAME_SUFFIX:=
+GENERIC_BACKPORT_DIR := $(GENERIC_PLATFORM_DIR)/backport$(if $(wildcard $(GENERIC_PLATFORM_DIR)/backport-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
+GENERIC_PATCH_DIR := $(GENERIC_PLATFORM_DIR)/pending$(if $(wildcard $(GENERIC_PLATFORM_DIR)/pending-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
+GENERIC_HACK_DIR := $(GENERIC_PLATFORM_DIR)/hack$(if $(wildcard $(GENERIC_PLATFORM_DIR)/hack-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
+GENERIC_FILES_DIR := $(foreach dir,$(wildcard $(GENERIC_PLATFORM_DIR)/files $(GENERIC_PLATFORM_DIR)/files-$(KERNEL_PATCHVER)),"$(dir)")
+$(warning GENERIC_BACKPORT_DIR = $(GENERIC_BACKPORT_DIR))
+include $(INCLUDE_DIR)/kernel-version.mk
+include $(INCLUDE_DIR)/kernel.mk
+
 PKG_NAME:=linux
 PKG_VERSION:=$(LINUX_VERSION)
 PKG_SOURCE:=$(LINUX_SOURCE)
@@ -36,7 +47,7 @@ else
   PKG_SOURCE:=$(LINUX_SOURCE)
   PKG_SOURCE_URL:=$(LINUX_SITE)
 endif
-HOST_BUILD_DIR:=$(KERNEL_BUILD_DIR)/linux-$(LINUX_VERSION)
+HOST_BUILD_DIR:=$(KERNEL_BUILD_DIR)/linux-$(LINUX_VERSION)$(KERNEL_NAME_SUFFIX)
 PKG_HASH:=$(LINUX_KERNEL_HASH)
 LINUX_DIR := $(HOST_BUILD_DIR)
 FILES_DIR := 
-- 
2.34.1

